node/core/candidate-validation/src/lib.rs

```diff
@@ -362,22 +373,30 @@ async fn validate_candidate_exhaustive(
 		&descriptor,
 		persisted_validation_data.max_pov_size,
 		&*pov,
-		&validation_code_hash,
+		validation_code_hash.as_ref(),
 	) {
 		return Ok(Ok(ValidationResult::Invalid(e)))
 	}
 
-	let raw_validation_code = match sp_maybe_compressed_blob::decompress(
-		&validation_code.0,
-		VALIDATION_CODE_BOMB_LIMIT,
-	) {
-		Ok(code) => code,
-		Err(e) => {
-			tracing::debug!(target: LOG_TARGET, err=?e, "Invalid validation code");
+	let validation_code = if let Some(validation_code) = validation_code {
+		let raw_code = match sp_maybe_compressed_blob::decompress(
+			&validation_code.0,
+			VALIDATION_CODE_BOMB_LIMIT,
+		) {
+			Ok(code) => code,
+			Err(e) => {
+				tracing::debug!(target: LOG_TARGET, err=?e, "Invalid validation code");
```

Comment:
This message is a bit too broad. Maybe we can reuse the error here, like "code decompression failure"?